src/builder.js

```diff
@@ -25,40 +22,40 @@ class JUnitReportBuilder {
   }
 
   /**
-   * @returns {string} xml file content
+   * @returns {string}
    */
   build() {
-    var xmlTree = xmlBuilder.create('testsuites', { encoding: 'UTF-8', invalidCharReplacement: '' });
-    _.forEach(this._testSuitesAndCases, function (suiteOrCase) {
-      suiteOrCase.build(xmlTree);
-    });
+    var xmlTree = this._rootTestSuites.build();
     return xmlTree.end({ pretty: true });
   }
 
   /**
-   * @returns {import('./test_suite')}
+   * @returns {import('./test_suites').TestSuites}
+   */
+  testSuites() {
+    return this._rootTestSuites;
+  }
+
+  /**
+   * @returns {import('./test_suite').TestSuite}
    */
   testSuite() {
-    var suite = this._factory.newTestSuite();
-    this._testSuitesAndCases.push(suite);
-    return suite;
+    return this._rootTestSuites.testSuite();
   }
 
   /**
-   * @returns {import('./test_case')}
+   * @returns {import('./test_case').TestCase}
    */
   testCase() {
-    var testCase = this._factory.newTestCase();
-    this._testSuitesAndCases.push(testCase);
-    return testCase;
+    return this._rootTestSuites.testCase();
   }
 
   /**
-   * @returns {ReturnType<import('./factory')['newBuilder']>}
+   * @returns {JUnitReportBuilder}
    */
   newBuilder() {
-    return this._factory.newBuilder();
+    return new JUnitReportBuilder(this._factory);
   }
 }
 
-module.exports = JUnitReportBuilder;
+module.exports = { Builder: JUnitReportBuilder };
```

Comment:
Outside of `src/index.js` this won't have any affect as `require('junit-report-builder')` get's the default export of that file, not these files I refactored, because it resolves to the main field in `package.json`.

Technically they are breaking changes if someone does `require('junit-report-builder/src/builder')` then it will break, but technically we're not supposed to do that with node_modules 😂.

That said, this is a habit of mine - doing exports like this makes sure that we use the same name everywhere or explicitly rename it.